src/core/tests/pass/serialization/serialize.cpp

```diff
@@ -51,6 +53,32 @@ TEST_P(SerializationTest, CompareFunctions) {
     EXPECT_TRUE(res.valid) << res.message;
 }
 
+TEST(SerializeCovolutionTest, CheckPads) {
+    auto input = std::make_shared<ov::opset8::Parameter>(ov::element::f16, ov::Shape{1, 3, 12, 12});
+    auto const_weights =
+        ov::opset8::Constant::create(ov::element::f16, ov::Shape{1, 3, 3, 3}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 2, 3, 4, 5,
+                                                                               6, 7, 8, 9, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+    auto conv = std::make_shared<ov::opset8::Convolution>(input,
+                                                          const_weights,
+                                                          ov::Strides{1, 1},
+                                                          ov::CoordinateDiff{1, 1},
+                                                          ov::CoordinateDiff{1, 1},
+                                                          ov::Strides{1, 1},
+                                                          ngraph::op::PadType::AUTO);
+    auto f = std::make_shared<ov::Model>(ov::NodeVector{conv}, ov::ParameterVector{input});
+
+    ov::pass::Manager manager;
+    manager.register_pass<ngraph::pass::Serialize>("pads_test.xml", "pads_test.bin");
+    ASSERT_NO_THROW(manager.run_passes(f));
+
+    auto pad = ov::CoordinateDiff{1, 1};
+    ASSERT_EQ(conv->get_pads_begin(), pad);
+    ASSERT_EQ(conv->get_pads_end(), pad);
```

Comment:
It is better to use `CompareFunctions` here. So you first copy function, then call Serialize and then compare functions.
Also I would suggest to extend other existing tests with this check.